src/bundle/DependencyInjection/ContainerBuilder.php

```diff
@@ -59,4 +64,73 @@ private function resolveExtensionName(string $name): string
 
         return self::$extensionNameMap[$name] ?? $name;
     }
+
+    public function getDefinition(string $id)
+    {
+        if (!isset(self::$serviceNameMap)) {
+            /** @noinspection PhpIncludeInspection */
+            self::$serviceNameMap = require IbexaCompatibilityLayerBundle::MAPPINGS_PATH
+                . \DIRECTORY_SEPARATOR . 'services-to-fqcn-map.php';
+        }
+
+        try {
+            return parent::getDefinition($id);
+        } catch (ServiceNotFoundException $e) {
+            if (isset(self::$serviceNameMap[$id])) {
+                return parent::getDefinition(self::$serviceNameMap[$id]);
+            }
+        }
+
+        throw $e;
```

Comment:
Hmm, I'm not a huge fan of referencing variable set outside of the scope, though in this case I guess it will always work. I would rather check for the existence of the service in the map and always substitute rather than relying on exceptions.